### PR TITLE
Using PistonWindow + gfx::Canvas [WIP]

### PR DESCRIPTION
**Pull request by stjahns**

### CODE REVIEW

**Comment by stjahns** [comment]
Closes https://github.com/PistonDevelopers/gfx-debug-draw/issues/23

File: src/debug_renderer.rs [hunk 1]
```diff
@@ -25,6 +25,28 @@ pub struct DebugRenderer<R: gfx::Resources> {
...
+        D: Device<Resources = R, CommandBuffer = C>,
+    > (
+        canvas: &mut gfx::Canvas<O, D, F>,
+        frame_size: [u32; 2],
+        initial_buffer_size: usize,
+        bitmap_font: Option<BitmapFont>,
+        bitmap_font_texture: Option<gfx::TextureHandle<R>>,
+    ) -> Result<DebugRenderer<R>, DebugRendererError> {
+
+        let &mut gfx::Canvas {
```
Comment (kvark) [inline src/debug_renderer.rs:41]: no need for that, just use `&mut canvas.device` and such where you need to

File: src/debug_renderer.rs [hunk 2]
```diff
@@ -87,6 +109,25 @@ impl<R: gfx::Resources> DebugRenderer<R> {
...
+        C: gfx::CommandBuffer<R>,
+        F: Factory<R>,
+        O: gfx::render::target::Output<R>,
+        D: Device<Resources = R, CommandBuffer = C>,
+    > (
+        &mut self,
+        canvas: &mut gfx::Canvas<O, D, F>,
+        projection: [[f32; 4]; 4],
+    ) {
+        let &mut gfx::Canvas {
```
Comment (kvark) [inline src/debug_renderer.rs:122]: same here

File: src/debug_renderer.rs [hunk 3]
```diff
@@ -25,6 +25,28 @@ pub struct DebugRenderer<R: gfx::Resources> {
...
 impl<R: gfx::Resources> DebugRenderer<R> {
 
+    pub fn from_canvas<
+        C: gfx::CommandBuffer<R>,
+        F: Factory<R>,
+        O: gfx::render::target::Output<R>,
+        D: Device<Resources = R, CommandBuffer = C>,
+    > (
+        canvas: &mut gfx::Canvas<O, D, F>,
+        frame_size: [u32; 2],
```
Comment (kvark) [inline src/debug_renderer.rs:35]: no need for the size either. `Canvas::output` implements `gfx::Output`, which you can ask for `.get_size()`

**Pull request merged**